### PR TITLE
[docs] Improve distributed tracing documentation

### PR DESCRIPTION
**Pull request by bmorelli25**
Closes https://github.com/elastic/apm-agent-rum-js/issues/364.

This PR cleans up some grammar and adds headings for content and code snippets in the distributed tracing guide. It also adds links to the relevant Agent API documentation for dynamically generated HTML.

### CODE REVIEW

File: docs/distributed-tracing-guide.asciidoc [hunk 1]
```diff
@@ -23,22 +26,23 @@ var apm = initApm({
...
-
 This effectively tells the agent to add the distributed tracing HTTP header (`elastic-apm-traceparent`) 
 to requests made to `https://api.example.com`.
 
+To read more about cross-origin requests and why this process is necessary,
+please see the MDN page on https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS[Cross-Origin Resource Sharing].
+
 [float]
 [[server-configuration]]
 === Server configuration
```
Comment (bmorelli25) [inline docs/distributed-tracing-guide.asciidoc]: I'm confused about this heading. It seems the content both before and after the heading relate to CORS?

File: docs/distributed-tracing-guide.asciidoc [hunk 2]
```diff
@@ -23,22 +26,23 @@ var apm = initApm({
 })
 ----
 
-
 This effectively tells the agent to add the distributed tracing HTTP header (`elastic-apm-traceparent`) 
 to requests made to `https://api.example.com`.
 
+To read more about cross-origin requests and why this process is necessary,
```
Comment (vigneshshanmugam) [inline docs/distributed-tracing-guide.asciidoc]: Can we also mentioned here that distributed tracing headers would be only appended to `API` calls? so it becomes clear and we can link to the below dynamically generated html heading incase users wants to view the full trace?

File: docs/distributed-tracing-guide.asciidoc (hunk 1 above)
Comment (jahtalab) [reply docs/distributed-tracing-guide.asciidoc]: The part before this is related to what needs to be done on the frontend side (e.g. configuring the agent), but the server configuration refers to what needs to be done in the backend side. They are part of the CORS configuration just different places.

Comment (bmorelli25) [reply docs/distributed-tracing-guide.asciidoc]: Got it, thank you!

File: docs/distributed-tracing-guide.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/distributed-tracing-guide.asciidoc]: Can you check the latest commit and let me know if that covers it?

Comment (vigneshshanmugam) [reply docs/distributed-tracing-guide.asciidoc]: Yes looks good Brandon.

**Comment by jahtalab** [comment]
The test failures are unrelated to these changes.

**Pull request merged**